lib/join/index.js

```diff
@@ -0,0 +1,5 @@
+import curry from '../curry'
+
+export default curry(function join(d, arr) {
```

Comment:
curryN(2, ...) will take less memory
and check the size adding join and curry at the same file if it's worth it (i still don't know)